### ISSUE TITLE
Generics Regression in 1.1.373

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#8552]
Starting in 1.1.373, an example I had like this starts to fail whenever T isn't contravariant:

**Passes**

```python
from collections.abc import Generator
from typing import NamedTuple, Generic, TypeVar

T = TypeVar("T", contravariant=True)


class Container(NamedTuple, Generic[T]):
    value: T


class Source(Generic[T]):
    def ingest(self) -> Generator[Container[T], None, None]: ...


class PrimarySource(Source[T]):
    def __init__(self, obj: T) -> None:
        self.obj = obj

    def ingest(self) -> Generator[Container[T], None, None]:
        yield Container(self.obj)
```

**Fails**

```python
from collections.abc import Generator
from typing import NamedTuple, Generic, TypeVar

T = TypeVar("T")


class Container(NamedTuple, Generic[T]):
    value: T


class Source(Generic[T]):
    def ingest(self) -> Generator[Container[T], None, None]: ...


class PrimarySource(Source[T]):
    def __init__(self, obj: T) -> None:
        self.obj = obj

    def ingest(self) -> Generator[Container[T], None, None]:
        yield Container(self.obj)
```

[Pyright Playground
](https://pyright-play.net/?pyrightVersion=1.1.373&strict=true&code=FAMwTg9gtgBAxhANogpnALgSwgOwM4B0AhgEZwyZQAOEY6MA4ijimEeraJLOgJ5WYcAcwrVa9AHJEoKACYAVAK5VUAGkbNWmOOvn8UANSJhgweQH0EOdGxgBeGHqqHjACgBEFqzaLv13tgA3Y0wiazt5MEUUAEpTYDhEIjw8GABhXHQiQVZXKRkFZTUNFjBtAG0vTLYAXRiALmAYZphgxGj6x0tqonjE5NSAZQhFMDgUVyZSiqrrWoamltkUEAphFDx0VzwURBAYmABaAD4S1nZacozrbNLK7rmiGvUJXBQXt5rOgh%2B%2BpJSYAAFMpQYy8YajcauCFjFD3AJPBYtGDLVbmcyCTDodHbXYgdQQEgAK06sx8BxOMFeLEayOROz2BEJRPsMGZpmRqLWQg2WwZ%2ByOpym5w4YCumVurHhPWeVLeHxYX0WdJgvEwu1k6QlOTAuMZzLiQA)

I don't see any reason here as for why T would need to be contravariant for this example to pass, but if this actually was a false negative in the past, feel free to close this out.

**Comment by erictraut** [comment]
Thanks for reporting. This is definitely a regression. It will be fixed in the next release.

**Comment by max-muoto** [comment]
> Thanks for reporting. This is definitely a regression. It will be fixed in the next release.

Thanks!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.374.
